File: 4 array/3 selection sort/selection.js
const selectionSort = (arr) => {
    size = arr.length;
    for (let i = 0; i < size - 1; i++) {
        let smalIndex = i;

        for (let j = i + 1; j < size; j++) {
            if (arr[smalIndex] > arr[j]) {
                smalIndex = j;
            }
        }
        // swap
        const temp = arr[smalIndex];
        arr[smalIndex] = arr[i];
        arr[i] = temp;
    }

    console.log(`Selection sort array is ${arr}`);
}

const array = [7, 8, 4, 9, 3];

console.log(`Original array is ${array}`);

selectionSort(array)

File: 4 array/3 selection sort/Selection.java
// Selection sort algorithm in java

public class Selection {
    // sorting fnction
    static void selectionSort(int arr[], int size) {
        for (int i = 0; i < size - 1; i++) {
            int smalIndex = i;

            for (int j = i + 1; j < size; j++) {
                if (arr[smalIndex] > arr[j]) {
                    smalIndex = j;
                }
            }
            // swap
            int temp = arr[smalIndex];
            arr[smalIndex] = arr[i];
            arr[i] = temp;
        }

        // to print the sorted array
        System.out.print("Selection sort array is : ");
        for (int i : arr) {
            System.err.print(i + " ");
        }
    }

    // Main Function
    public static void main(String[] args) {
        int array[] = { 7, 8, 4, 9, 3 };
        int size = array.length;

        System.out.print("Orignal array is : ");
        for (int i : array) {
            System.out.print(i + " ");
        }
        System.out.println();

        selectionSort(array, size);
    }
}
